pkg/prometheus/promcfg.go

```diff
@@ -3320,6 +3320,72 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	//KumaSDConfig
+	if len(sc.Spec.KumaSDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.KumaSDConfigs))
+		for i, config := range sc.Spec.KumaSDConfigs {
+			assetStoreKey := fmt.Sprintf("scrapeconfig/%s/%s/kumasdconfig/%d", sc.GetNamespace(), sc.GetName(), i)
+			configs[i] = cg.addBasicAuthToYaml(configs[i], assetStoreKey, store, config.BasicAuth)
+			configs[i] = cg.addSafeAuthorizationToYaml(configs[i], fmt.Sprintf("scrapeconfig/auth/%s/%s/kumasdconfig/%d", sc.GetNamespace(), sc.GetName(), i), store, config.Authorization)
+			configs[i] = cg.addOAuth2ToYaml(configs[i], config.OAuth2, store.OAuth2Assets, assetStoreKey)
+			configs[i] = cg.addProxyConfigtoYaml(ctx, configs[i], sc.GetNamespace(), store, config.ProxyConfig)
+
+			if config.Server != "" {
```

Comment:
`Server` is a required field right, we don't need to check for empty here

Reply:
Done